addons/event/models/event_event.py

```diff
@@ -286,17 +284,19 @@ def _compute_start_sale_date(self):
             start_dates = [ticket.start_sale_datetime for ticket in event.event_ticket_ids if not ticket.is_expired]
             event.start_sale_datetime = min(start_dates) if start_dates and all(start_dates) else False
 
-    @api.depends('event_ticket_ids.sale_available')
+    @api.depends('event_ticket_ids.sale_available', 'seats_available', 'seats_limited')
     def _compute_event_registrations_sold_out(self):
+        """Note that max seats limits for events and sum of limits for all its tickets may not be
+        equal to enable flexibility.
+        E.g. max 20 seats for ticket A, 20 seats for ticket B
+            * With max 20 seats for the event
+            * Without limit set on the event (=40, but the customer didn't explicitly write 40)
+        """
         for event in self:
-            if event.seats_limited and not event.seats_available:
-                event.event_registrations_sold_out = True
-            elif event.event_ticket_ids:
-                event.event_registrations_sold_out = not any(
-                    ticket.seats_available > 0 if ticket.seats_limited else True for ticket in event.event_ticket_ids
-                )
-            else:
-                event.event_registrations_sold_out = False
+            event.event_registrations_sold_out = (
+                event.seats_limited and not event.seats_available
+                or all(map(lambda x: x.is_sold_out, event.event_ticket_ids))
```

Comment:
Maybe we could construct an or-based condition, allowing to get out of the loop as soon as we have a greened condition (using an any for tickets), like
```
event.event_registrations_sold_out =
    not event.seats_limited or \
    not event.seats_available or \
    any(not ticket.is_sold_out for ticket in event.event_ticket_ids) if event.event_ticket_ids else True
```

Reply:
Don't we already have the same early stopping behavior? `and` and `or` do that, and `all` too returns `False` as soon as one `False` value is met, per documentation https://docs.python.org/3/library/functions.html#all

But we do need to have a generator for the lazy evaluation, so I updated that. Thanks!
(ref: https://stackoverflow.com/questions/64090762/python-lazy-function-evaluation-in-any-all).

As you taught me, `is_sold_out` will be `False` if `event_ticket_ids` is `False`, so we can avoid that additional `ìf`, right?
Finally, I also used the `\` you prefer :-)

Reply:
Mmmh, as ``all([])`` is True you will have "or True", which means that without tickets this is never sold out ?

Reply:
You're right again. This should do it, while we discuss about the logic.